pilot/cmd/istioctl/inject.go

```diff
@@ -18,32 +18,64 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"os"
+
 	// TODO(nmittler): Remove this
+	"github.com/ghodss/yaml"
 	_ "github.com/golang/glog"
 	"github.com/spf13/cobra"
-	"k8s.io/api/core/v1"
 
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/cmd"
+	"istio.io/istio/pilot/model"
 	"istio.io/istio/pilot/platform/kube"
 	"istio.io/istio/pilot/platform/kube/inject"
 	"istio.io/istio/pkg/log"
 	"istio.io/istio/pkg/version"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const configMapKey = "mesh"
+
+func getMeshConfigFromConfigMap(kubeconfig string) (*meshconfig.MeshConfig, error) {
+	_, client, err := kube.CreateInterface(kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	config, err := client.CoreV1().ConfigMaps(istioNamespace).Get(meshConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not find %q ConfigMap in %q namespace", meshConfigMapName, istioNamespace)
+	}
+	// values in the data are strings, while proto might use a
+	// different data type.  therefore, we have to get a value by a
+	// key
+	yaml, exists := config.Data[configMapKey]
+	if !exists {
+		return nil, fmt.Errorf("missing configuration map key %q", configMapKey)
+	}
+	return model.ApplyMeshConfigDefaults(yaml)
+}
+
 var (
```

Comment:
ideally the diff would be minimized (ie don't move meshConfigMapName at the end / what does that buy?)

Reply:
`meshConfigMapName` makes more sense next to `meshConfigFile` since they serve the same purpose and are mutually exclusive. 

The move here came as a result of me first trying to remove ConfigMap dependency completely from `kube-inject` but then added it back after it proved too much of a pain while trying to fix the e2e smoke tests.